src/Arcus.Templates.AzureFunctions.ServiceBus.Topic/OrderFunction.cs

```diff
@@ -0,0 +1,56 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Arcus.Messaging.Abstractions;
+using Arcus.Messaging.Abstractions.ServiceBus;
+using Arcus.Messaging.Abstractions.ServiceBus.MessageHandling;
+using Arcus.Templates.AzureFunctions.ServiceBus.Topic.Model;
+using Azure.Messaging.ServiceBus;
+using GuardNet;
+using Microsoft.Azure.ServiceBus;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Extensions.Logging;
+
+namespace Arcus.Templates.AzureFunctions.ServiceBus.Topic
+{
+    /// <summary>
+    /// Represents an Azure Service Bus topic trigger that processes <see cref="Order"/> messages.
+    /// </summary>
+    public class OrderFunction
+    {
+        private readonly IAzureServiceBusMessageRouter _messageRouter;
+        private readonly string _jobId;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="OrderFunction" /> class.
+        /// </summary>
+        /// <param name="messageRouter">The message router instance to route the Azure Service Bus topic messages through the order processing.</param>
+        /// <exception cref="ArgumentNullException">Thrown when the <paramref name="messageRouter"/> is <c>null</c>.</exception>
+        public OrderFunction(IAzureServiceBusMessageRouter messageRouter)
+        {
+            Guard.NotNull(messageRouter, nameof(messageRouter), "Requires an message router instance to route the incoming Azure Service Bus topic message through the order processing");
+
+            _messageRouter = messageRouter;
+            _jobId = Guid.NewGuid().ToString();
+        }
+
+        /// <summary>
+        /// Process an Azure Service Bus topic <paramref name="message"/> as an <see cref="Order"/>.
+        /// </summary>
+        /// <param name="message">The incoming message on the Azure Service Bus topic, representing an <see cref="Order"/>.</param>
+        /// <param name="log">The logger instance to write informational messages during the message processing.</param>
+        /// <param name="cancellationToken">The token to cancel the message processing.</param>
+        [FunctionName("order-processing")]
+        public async Task Run(
+            [ServiceBusTrigger("order-topic", "order-subscription",  Connection = "ServiceBusConnectionString")] ServiceBusReceivedMessage message,
+            ILogger log,
```

Comment:
Should we create a template that receives one message by default, or multiple messages ?

Reply:
Not sure if it really matters here? What will usually be used? Batches? We should be careful indeed as to not force too much on people.

Reply:
Indeed , it all depends on the use case.
However, creating a function that receives multiple messages at once, is simply done by specifing that the input object is an array instead of a single message.  We can leave it as it is imho.